src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -176,6 +178,83 @@ protected void LookupMembersInternal(LookupResult result, NamespaceOrTypeSymbol
             }
         }
 
+#nullable enable
+        protected void LookupImplicitExtensionMembersInSingleBinder(LookupResult result, Binder binder, TypeSymbol type,
+            string name, int arity, ConsList<TypeSymbol> basesBeingResolved, LookupOptions options,
+            Binder originalBinder, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
+        {
+            var compatibleExtensions = ArrayBuilder<TypeSymbol>.GetInstance();
+            getCompatibleExtensions(binder, type, compatibleExtensions);
+            // PROTOTYPE test use-site diagnostics
+            var tempResult = LookupResult.GetInstance();
+            foreach (TypeSymbol extension in compatibleExtensions)
+            {
+                // No need for "diagnose" since we discard the lookup results (and associated diagnostic info)
+                // unless the results are good.
+                LookupMembersWithoutInheritance(tempResult, extension, name, arity, options, originalBinder,
+                    accessThroughType: type, diagnose: false, ref useSiteInfo, basesBeingResolved);
+
+                MergeHidingLookupResults(result, tempResult, basesBeingResolved, ref useSiteInfo);
+                tempResult.Clear();
+            }
+
+            tempResult.Free();
+            compatibleExtensions.Free();
+            return;
+
+            void getCompatibleExtensions(Binder binder, TypeSymbol type, ArrayBuilder<TypeSymbol> compatibleExtensions)
+            {
+                var extensions = ArrayBuilder<TypeSymbol>.GetInstance();
+                binder.GetImplicitExtensionTypes(extensions, originalBinder: this);
+
+                foreach (var extension in extensions)
+                {
+                    // PROTOTYPE deal with generic extensions
+                    if (isCompatible(extension, type))
+                    {
+                        compatibleExtensions.Add(extension);
+                    }
+                }
+
+                extensions.Free();
+            }
+
+            bool isCompatible(TypeSymbol extension, TypeSymbol type)
```

Comment:
>type [](http://example.com/codeflow?start=63&length=4)

Are we supposed to handle type parameters here?

Reply:
No. I'll tighten with assertions. 

The spec for "member access" says:
> - (only relevant in phase B) If `E` is a property access, indexer access, variable, or value, 
  the type of which is `T`, **where `T` is not a type parameter**, and 
  an **extension member lookup** of `I` in `T` with `K` type arguments produces a match, 
  then `E.I` is evaluated and classified as follows:

> - If `E` is classified as a type, **if `E` is not a type parameter**, 
  and if an extension member lookup of `I` in `E` with `K` type parameters produces a match, 
  then `E.I` is evaluated and classified as follows: